internal/storage/gcs/bucket.go

```diff
@@ -20,6 +20,39 @@ import (
 	"golang.org/x/net/context"
 )
 
+// ChunkUploader interface represents asynchronous object
+// uploaders which upload a chunk of an object's data in a
+// call using GCS' resumable upload API.
+//
+// On closing, they return a GCS object (gcs.Object)
+// and and error object for error-handling.
+type ChunkUploader interface {
+	// UploadChunkAsync writes the entire input content by reading
```

Comment:
UploadChunkAsync uploads the given chunk to gcs. Progress should be tracked using the progress func passed during ChunkUploader instance creation.

Reply:
Changed to your write-up.